inst/scripts/importPDB.R

```diff
@@ -0,0 +1,158 @@
+    # This element will be removed from the list of genes at the end of each
+    # iteration of the while loop.This is necessary for pausing the process and
+    # continuing it later.
+
+    curr_gene <- genes[[1]]
+
+    # Translate the cDNA of the gene to its aa sequence
+    targetAA <- AAString(paste(a.a. sequence of the gene, collapse = "")
+
+    for (pdbID_chainID matched to gene){
+        pdbID <- pdbid retrieved from the MT object
+        chainID <- chainid retrieved from the MT object
+
+        #=====================READ SUBJECT PDB FILE============================
+
+        # download the PDB file.
+        subject <- read.pdb(pdbID)
+        # extract the desired chain and get its aa sequence.
+        chainAAIndices <- atom.select(subject,
+                                      "calpha",
+                                      chain = chainID,
+                                      value = TRUE)
+        subjectAA <- AAString(paste(aa321(chainAAIndices$atom$resid), collapse = ""))
+
+        #=====================ALIGNMENT========================================
+
+        # retrieve the start and end indices for the target & subject sequences:
+        start_target <- the starting index of the alignment in targetAA from MT
+        end_target <- the last index of the alignment in targetAA from MT
+        start_subject <- the starting index of the alignment subjectAA from MT
+        end_subject <- the last index of the alignment in target AA from MT
+
+        # perform pairwise alignment with default parameters.
+        ali <- pairwiseAlignment(subjectAA[start_subject, end_subject],
+                                 targetAA[start_target, end_target])
+
+        #======================================================================
+
+        # create a data frame with nmatch(ali) rows.
+        DF <- data.frame("ResID" = c(1:nmatch(ali)),
+                         "AA Type" = c(1:nmatch(ali)),
+                         "Sidechain Centroid" = c(1:nmatch(ali)),
+                         "C.alpha Centroid" = c(1:nmatch(ali)))
+
+        # for all matched residues:
+        for (n in 1:nmatch(ali)){
+            i <- index of residue in targetAA
+            j <- index of residue in subjectAA
+
+            # rowname will be the index of the residue in target gene:
+            rownames(DF)[n] <- i
+
+            #=================RESIDUE ID & AA TYPE=============================
+
+            # resids are the 3 letter code of the aa:
+            resID <- unique(subject$atom[subject$atom$resno==j,]$resid)
+            # type of AA..
+            aaType <- type of targetAA[i]
+            # ^^^ Q: what does it mean by amino acid type?
+            #        (hydrophobic vs. hydrophilic? or acidic vs. basic? or
+            #        something else?) Still need to figure out why we need
+            #        these.. They are mentioned in the design task page of
+            #        the GeneData object.

```

Comment:
> One of the twenty one-letter amino acid codes.
Is that the resID or aaType?

then what would the other one be? because I already have resID as the 3-letter amino acid code.